compiler/rustc_const_eval/src/interpret/memory.rs

```diff
@@ -277,6 +278,7 @@ impl<'tcx, M: Machine<'tcx>> InterpCx<'tcx, M> {
         new_size: Size,
         new_align: Align,
         kind: MemoryKind<M::MemoryKind>,
+        init: AllocInit,
```

Comment:
Might be good to explain in a doc comment for the fn that this only refers to the new, grown part of the alloc; the carried-over part is as (un)initialized as the old allocation.